lvsfunc/comparison.py

```diff
@@ -73,85 +387,91 @@ def _resample(clip: vs.VideoNode) -> vs.VideoNode:
-    :param warn:              Warns if the length of given clips don't align (Default: True)
-
-    :return:                  Clip with clips stacked
-    """
-    if len(clips) < 2:
-        raise ValueError(f"stack_compare: 'Too few clips supplied'")
-
-    if len(clips) != 2 and make_diff:
-        raise ValueError(f"stack_compare: 'You can only create a diff for two clips'")
-
-    formats = set()
-    for c in clips:
-        if c.format is None:
-            raise ValueError("stack_compare: 'Variable-format clips not supported'")
-        formats.add(c.format.id)
-    if len(formats) != 1:
-        raise ValueError(f"stack_compare: 'The format of every clip must be equal'")
-
-    if make_diff:
-        diff = core.std.MakeDiff(clips[0], clips[1])
-        diff = core.resize.Spline36(diff, get_w(576), 576).text.FrameNum(8)
-        resize = [core.resize.Spline36(c, int(diff.width / 2), int(diff.height / 2)) for c in clips]
-        resize[0], resize[1] = resize[0].text.Text("Clip A", 3), resize[1].text.Text("Clip B", 1)
-        stack = core.std.StackVertical([core.std.StackHorizontal([resize[0], resize[1]]), diff])
-    else:
-        stack = core.std.StackHorizontal(clips)
-    if warn:
-        if len(set([c.num_frames for c in clips])) != 1:
-            stack = core.text.Text(stack, f"Clip Length Mismatch Detected! \nPlease make sure the lengths of all clips match!\n"+"".join(f"\nClip {i+1}: {c.num_frames} Frames" for i, c in enumerate(clips)), 2)
-    return stack
+    :param clips:     Clips to compare
+    :param make_diff: Create and stack a diff (only works if two clips are given) (Default: True)
+    :param height:    Height in px to rescale clips to if `make_diff` is ``True`` (MakeDiff clip will be twice this resolution) (Default: 288)
+    :param warn:      Unused parameter kept for backward compatibility
+
+    :return:          Clip with clips stacked
+    """
+    if not make_diff:
+        warnings.warn('stack_compare has been deprecated in favor of `lvsfunc.comparison.Stack` if not using `make_diff`', DeprecationWarning)
+        return core.std.StackHorizontal(clips)
+
+    if len(clips) != 2:
+        raise ValueError("stack_compare: `make_diff` only works for exactly 2 clips")
 
+    diff = core.std.MakeDiff(clips)
+    diff = diff.resize.Spline36(width=get_w(height, only_even=False) * 2, height=height * 2).text.FrameNum(8)
+    resized = [clips[0].resize.Spline36(width=get_w(height, only_even=False), height=height).text.Text('Clip A', 3),
+               clips[1].resize.Spline36(width=get_w(height, only_even=False), height=height).text.Text('Clip B', 1)]
 
-def stack_planes(clip: vs.VideoNode,
-                 stack_vertical: bool = False) -> vs.VideoNode:
+    return Stack(direction=Direction.VERTICAL,
+                 **{'': Stack(direction=Direction.HORIZONTAL,
+                              **{'': resized[0], ' ': resized[1]}
```

Comment:
See how awkward this `**` stuff is? 😛 

In fact, seeing how it is used here, a list or generic iterable should also be accepted instead of only a dict.